sdk/storage/azblob/container/client.go

```diff
@@ -308,3 +313,42 @@ func (c *Client) GetSASURL(permissions sas.ContainerPermissions, start time.Time
 
 	return endpoint, nil
 }
+
+// GetUDKSASURL is a convenience method for generating a SAS token for the currently pointed at container.
+// It can only be used if the credential supplied during creation was a UserDelegationKey.
+func (c *Client) GetUDKSASURL(permissions sas.ContainerPermissions, start time.Time, expiry time.Time) (string, error) {
+	if c.userDelegationKey() == nil {
+		return "", errors.New("GetUDKSASURL can only be signed with a UserDelegationKey")
+	}
+
+	urlParts, err := blob.ParseURL(c.URL())
+	if err != nil {
+		return "", err
+	}
+
+	udc := azblob.UserDelegationCredential{
```

Comment:
Please use the source from `exported`.  We don't want to add a dependency on the `azblob` package.